src/main/js/store/actions/globalConfiguration.js

```diff
@@ -258,17 +303,29 @@ export function updateConfig(config) {
             request = ConfigRequestBuilder.createNewConfigurationRequest(ConfigRequestBuilder.CONFIG_API_URL, csrfToken, config);
         }
         request.then((response) => {
-            response.json()
-                .then((responseData) => {
-                    if (response.ok) {
-                        const newId = responseData.id;
-                        const updatedConfig = FieldModelUtilities.updateFieldModelSingleValue(config, 'id', newId);
-                        dispatch(configUpdated(updatedConfig));
-                        dispatch(refreshConfig(newId));
-                    } else {
-                        handleFailureResponse(dispatch, responseData, response.status);
-                    }
-                });
+            if (response.ok) {
+                if (response.status === 201) {
+                    response.json()
+                        .then((responseData) => {
+                            const newId = responseData.id;
+                            const updatedConfig = FieldModelUtilities.updateFieldModelSingleValue(config, 'id', newId);
+                            dispatch(configUpdated(updatedConfig));
+                            dispatch(refreshConfig(newId));
+                        });
+                } else {
+                    dispatch(refreshConfig(id));
+                }
```

Comment:
Would this work? Or is the `response.json()` promise preventing this?
```suggestion
                let newId = id;
                if (response.status === 201) {
                    response.json()
                        .then((responseData) => {
                           newId = responseData.id;
                            const updatedConfig = FieldModelUtilities.updateFieldModelSingleValue(config, 'id', newId);
                            dispatch(configUpdated(updatedConfig));
                        });
                }
                dispatch(refreshConfig(newId));
```

Reply:
You can't do your proposed change because the parsing of the responData happens in a .then.  It is a promise.  It executes at a different time than the dispatch(refreshConfig(newId) that is proposed on line 316.  It needs to be the way that it is written because of the promise that is processing the response body.